biodatasets/nlm_gene/nlm_gene.py

```diff
@@ -0,0 +1,316 @@
+                    {
+                        "passages": [
+                            {
+                                "document_id": datasets.Value("string"),
+                                "type": datasets.Value("string"),
+                                "text": datasets.Value("string"),
+                                "entities": [
+                                    {
+                                        "id": datasets.Value("string"),
+                                        "offsets": [[datasets.Value("int32")]],
+                                        "text": [datasets.Value("string")],
+                                        "type": datasets.Value("string"),
+                                        "normalized": [
+                                            {
+                                                "db_name": datasets.Value("string"),
+                                                "db_id": datasets.Value("string"),
+                                            }
+                                        ],
+                                    }
+                                ],
+                                "relations": [
+                                    {
+                                        "id": datasets.Value("string"),
+                                        "type": datasets.Value("string"),
+                                        "arg1_id": datasets.Value("string"),
+                                        "arg2_id": datasets.Value("string"),
+                                    }
+                                ],
+                            }
+                        ]
+                    }
+                )
+
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[self.config.schema]
+        data_dir = Path(dl_manager.download_and_extract(urls))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_dir/"Corpus",
+                    "file_name": "Pmidlist.Train.txt",
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": data_dir/"Corpus",
+                    "file_name": "Pmidlist.Test.txt",
+                    "split": "test",
+                },
+            ),
+        ]
+
+    @staticmethod
+    def _get_bioc_entity(span, doc_text, db_id_key="NCBI Gene identifier") -> dict:
```

Comment:
This is a very nasty dataset, which requires quite some quality checks. We don't aim to perfection but the dataset should give the correct entities at least. Please always verify that the entities are parsed correctly.
 In this case you can do:
```python
  for f in files:
  c = bioc.load(f)
  for d in c.documents:
    for p in d.passages:
      for a in p.annotations:
        print(a.infons.get("NCBI Gene identifier"))
)
```
This will show that there are multiple connectors for the normalization, e.g. `,`, `;`, `|`